src/main/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistency.java

```diff
@@ -39,12 +40,16 @@ public void write(Connection main) {
     public boolean isConsistent(Supplier<Connection> replicaSupplier) {
         Collection<Database> replicas = cluster.getReplicas(replicaSupplier);
 
-        if (replicas.isEmpty()) {
-            return false;
-        }
-
         return replicas.stream()
-            .allMatch(replica -> replicaConsistency.isConsistent(() -> replica.getConnectionSupplier().get()));
+            .map(replica -> {
+                try (Connection connection = replica.getConnectionSupplier().get()) {
+                    return replicaConsistency.isConsistent(() -> connection);
+                } catch (SQLException exception) {
+                    return false;
+                }
+            })
+            .reduce(Boolean::logicalAnd)
+            .orElse(false);
```

Comment:
if we want to autoclose obtained connections, then we cannot use `allMatch` as it is a short-circuit operator

Reply:
Can you elaborate?
AllMatch will not obtain all connections, but should close all the obtained connections.

Reply:
you're right, will refactor this